### PR TITLE
#5091 - Init racc/cparse

### LINKED ISSUE #5091: bundle path issue with racc - preventing to use openstudio extension gem's openstudio:test_with_openstudio.
Issue overview
--------------

We have gems that indirectly depend on racc. I can't seem to run with a bundle / Gemfile that has racc as a dependency on either of my machines, **but it works on nrel/openstudio:3.7.0 docker**, why?

rubocop 1.15.0 (and nokogiri and rubyXL) depend on racc.

example

https://github.com/NREL/OpenStudio-measure-tester-gem/blob/14659ef904019789f651a890bc56d2324d725805/openstudio_measure_tester.gemspec#L37

## Current Behavior

```
echo "source 'https://rubygems.org'" > Gemfile && echo "gem 'racc', '1.7.2'" >> Gemfile
bundle install --path=.bundle
openstudio --verbose --bundle Gemfile --bundle_path ./.bundle --bundle_without native_ext gem_list
```

```
$ openstudio classic --verbose --bundle Gemfile --bundle_path ./.bundle --bundle_without native_ext gem_list
D, [2024-02-15T21:51:25.543493 #58828] DEBUG -- : Adding dependency on racc '~> 1.7.2'

D, [2024-02-15T21:51:25.564884 #58828] DEBUG -- : Processing 61 activation requests
D, [2024-02-15T21:51:25.566216 #58828] DEBUG -- : Activating gem :/ruby/2.7.0/specifications/bundler-2.1.4.gemspec
I, [2024-02-15T21:51:25.968700 #58828]  INFO -- : without_groups = native_ext
I, [2024-02-15T21:51:25.968732 #58828]  INFO -- : g = default
I, [2024-02-15T21:51:25.968741 #58828]  INFO -- : Bundling with groups [default]
Error executing argv: ["--verbose", "--bundle", "Gemfile", "--bundle_path", "./.bundle", "--bundle_without", "native_ext", "gem_list"]
Error: Could not find racc-1.7.2 in any of the sources in :/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:86:in `block in materialize'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `map!'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `materialize'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/definition.rb:170:in `specs'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/definition.rb:237:in `specs_for'
:/openstudio_cli.rb:654:in `parse_main_args'
:/openstudio_cli.rb:784:in `execute'
:/openstudio_cli.rb:1987:in `<main>'
eval:193:in `eval'
eval:193:in `require_embedded_absolute'
eval:178:in `block in require_embedded'
eval:172:in `each'
eval:172:in `require_embedded'
eval:131:in `require'
eval:3:in `<main>'
```

## Steps to Reproduce

On my M1 mac and on my Ubuntu 20.04 it fails regardless of whether I try racc 1.7.2 (like the embedded one) or 1.7.3.  I use an RVM provided ruby on both, but I've tried manually unsetting GEM_HOME / GEM_PATH (even though the embedded ruby does that already)

On docker, `docker run -it --rm --platform linux/amd64 nrel/openstudio:3.7.0` it works with 1.7.2 and with 1.7..3

```
root@255a690713d7:/var/simdata/openstudio# openstudio --verbose --bundle Gemfile --bundle_path ./.bundle --bundle_without native_ext gem_list
Setting Log Level to Debug (-2)
[ApplicationPathHelpers] <-2> getOpenStudioModule, info.dli_fname = '/usr/local/openstudio-3.7.0/bin/../lib/libopenstudiolib.so'
[PathHelpers] <-2> completeAndNormalize: result = /usr/local/openstudio-3.7.0/lib/libopenstudiolib.so
[ApplicationPathHelpers] <-2> getOpenStudioModule, found '/usr/local/openstudio-3.7.0/lib/libopenstudiolib.so'
bundler (2.1.4) ':/ruby/2.7.0/gems/bundler-2.1.4'
racc (1.7.2) '/var/simdata/openstudio/.bundle/ruby/2.7.0/gems/racc-1.7.2'
```

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version):
 - Version of OpenStudio (if using an intermediate build, include SHA):

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #5091 
 - Companion PR: https://github.com/NREL/openstudio-gems/pull/73

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
Still doesn't fix the issue.

build_dir

```
mkdir test-bundle
cd test-bundle
echo "source 'https://rubygems.org'" > Gemfile && echo "gem 'racc', '1.7.3'" >> Gemfile
bundle install --path=.bundle
../Products/openstudio classic --verbose --bundle Gemfile --bundle_path $(pwd)/.bundle --bundle_without native_ext gem_list
┌────────────────────────────────────────────────────────────────────────────────┐
│  The `classic` command is deprecated and will be removed in a future release   │
└────────────────────────────────────────────────────────────────────────────────┘
I, [2024-02-21T01:33:55.450505 #23341]  INFO -- : CLI Parsed Inputs: ["--verbose", "--bundle", "Gemfile", "--bundle_path", "/Users/julien/Software/Others/OS-build-release/test-bundle/.bundle", "--bundle_without", "native_ext"] "gem_list" []
D, [2024-02-21T01:33:55.450538 #23341] DEBUG -- : Main arguments are ["--verbose", "--bundle", "Gemfile", "--bundle_path", "/Users/julien/Software/Others/OS-build-release/test-bundle/.bundle", "--bundle_without", "native_ext"]
D, [2024-02-21T01:33:55.450544 #23341] DEBUG -- : Sub-command is gem_list
D, [2024-02-21T01:33:55.450547 #23341] DEBUG -- : Sub-arguments are []
D, [2024-02-21T01:33:55.450558 #23341] DEBUG -- : Parsing main_args ["--verbose", "--bundle", "Gemfile", "--bundle_path", "/Users/julien/Software/Others/OS-build-release/test-bundle/.bundle", "--bundle_without", "native_ext"]
I, [2024-02-21T01:33:55.450571 #23341]  INFO -- : Setting BUNDLE_GEMFILE to Gemfile
I, [2024-02-21T01:33:55.450577 #23341]  INFO -- : Setting BUNDLE_PATH to /Users/julien/Software/Others/OS-build-release/test-bundle/.bundle
I, [2024-02-21T01:33:55.450582 #23341]  INFO -- : Setting BUNDLE_WITHOUT to native_ext
D, [2024-02-21T01:33:55.560697 #23341] DEBUG -- : Adding dependency on jaro_winkler '~> 1.5.4'
D, [2024-02-21T01:33:55.560727 #23341] DEBUG -- : Adding dependency on pycall '~> 1.2.1'
D, [2024-02-21T01:33:55.560735 #23341] DEBUG -- : Adding dependency on sqlite3 '~> 1.3.13.20180326210955'
D, [2024-02-21T01:33:55.560746 #23341] DEBUG -- : Adding dependency on tbd '~> 3.2.3'
D, [2024-02-21T01:33:55.560751 #23341] DEBUG -- : Adding dependency on addressable '~> 2.8.1'
D, [2024-02-21T01:33:55.560757 #23341] DEBUG -- : Adding dependency on ansi '~> 1.5.0'
D, [2024-02-21T01:33:55.560762 #23341] DEBUG -- : Adding dependency on ast '~> 2.4.2'
D, [2024-02-21T01:33:55.560768 #23341] DEBUG -- : Adding dependency on bcl '~> 0.7.1'
D, [2024-02-21T01:33:55.560773 #23341] DEBUG -- : Adding dependency on builder '~> 3.2.4'
D, [2024-02-21T01:33:55.560782 #23341] DEBUG -- : Adding dependency on bundler '~> 2.1.4'
D, [2024-02-21T01:33:55.560787 #23341] DEBUG -- : Adding dependency on docile '~> 1.4.0'
D, [2024-02-21T01:33:55.560793 #23341] DEBUG -- : Adding dependency on faraday '~> 1.0.1'
D, [2024-02-21T01:33:55.560800 #23341] DEBUG -- : Adding dependency on git '~> 1.12.0'
D, [2024-02-21T01:33:55.560806 #23341] DEBUG -- : Adding dependency on hana '~> 1.3.7'
D, [2024-02-21T01:33:55.560811 #23341] DEBUG -- : Adding dependency on json-schema '~> 2.7.0'
D, [2024-02-21T01:33:55.560819 #23341] DEBUG -- : Adding dependency on json_schemer '~> 2.0.0'
D, [2024-02-21T01:33:55.560825 #23341] DEBUG -- : Adding dependency on macaddr '~> 1.7.2'
D, [2024-02-21T01:33:55.560830 #23341] DEBUG -- : Adding dependency on matrix '~> 0.4.2'
D, [2024-02-21T01:33:55.560836 #23341] DEBUG -- : Adding dependency on minitar '~> 0.9'
D, [2024-02-21T01:33:55.560843 #23341] DEBUG -- : Adding dependency on minitest '~> 5.14.4'
D, [2024-02-21T01:33:55.560849 #23341] DEBUG -- : Adding dependency on minitest-reporters '~> 1.4.3'
D, [2024-02-21T01:33:55.560864 #23341] DEBUG -- : Adding dependency on msgpack '~> 1.4.2'
D, [2024-02-21T01:33:55.560870 #23341] DEBUG -- : Adding dependency on multipart-post '~> 2.4.0'
D, [2024-02-21T01:33:55.560877 #23341] DEBUG -- : Adding dependency on octokit '~> 4.18.0'
D, [2024-02-21T01:33:55.560884 #23341] DEBUG -- : Adding dependency on oga '~> 3.2'
D, [2024-02-21T01:33:55.560891 #23341] DEBUG -- : Adding dependency on openstudio-extension '~> 0.7.0'
D, [2024-02-21T01:33:55.560899 #23341] DEBUG -- : Adding dependency on openstudio-standards '~> 0.5.0'
D, [2024-02-21T01:33:55.560907 #23341] DEBUG -- : Adding dependency on openstudio-workflow '~> 2.3.1'
D, [2024-02-21T01:33:55.560914 #23341] DEBUG -- : Adding dependency on openstudio_measure_tester '~> 0.3.2'
D, [2024-02-21T01:33:55.560920 #23341] DEBUG -- : Adding dependency on oslg '~> 0.2.10'
D, [2024-02-21T01:33:55.561116 #23341] DEBUG -- : Adding dependency on osut '~> 0.3.0'
D, [2024-02-21T01:33:55.561124 #23341] DEBUG -- : Adding dependency on parallel '~> 1.19.1'
D, [2024-02-21T01:33:55.561130 #23341] DEBUG -- : Adding dependency on parser '~> 3.3.0.5'
D, [2024-02-21T01:33:55.561135 #23341] DEBUG -- : Adding dependency on public_suffix '~> 5.0.4'
D, [2024-02-21T01:33:55.561140 #23341] DEBUG -- : Adding dependency on racc '~> 1.7.3'
D, [2024-02-21T01:33:55.561145 #23341] DEBUG -- : Adding dependency on rainbow '~> 3.1.1'
D, [2024-02-21T01:33:55.561161 #23341] DEBUG -- : Adding dependency on rake '~> 13.1.0'
D, [2024-02-21T01:33:55.561167 #23341] DEBUG -- : Adding dependency on rchardet '~> 1.8.0'
D, [2024-02-21T01:33:55.561173 #23341] DEBUG -- : Adding dependency on regexp_parser '~> 2.9.0'
D, [2024-02-21T01:33:55.561179 #23341] DEBUG -- : Adding dependency on rexml '~> 3.2.5'
D, [2024-02-21T01:33:55.561184 #23341] DEBUG -- : Adding dependency on rubocop '~> 1.15.0'
D, [2024-02-21T01:33:55.561190 #23341] DEBUG -- : Adding dependency on rubocop-ast '~> 1.30.0'
D, [2024-02-21T01:33:55.561195 #23341] DEBUG -- : Adding dependency on rubocop-checkstyle_formatter '~> 0.4.0'
D, [2024-02-21T01:33:55.561209 #23341] DEBUG -- : Adding dependency on rubocop-performance '~> 1.11.5'
D, [2024-02-21T01:33:55.561215 #23341] DEBUG -- : Adding dependency on ruby-ll '~> 2.1.3'
D, [2024-02-21T01:33:55.561220 #23341] DEBUG -- : Adding dependency on ruby-ole '~> 1.2.12.2'
D, [2024-02-21T01:33:55.561228 #23341] DEBUG -- : Adding dependency on ruby-progressbar '~> 1.13.0'
D, [2024-02-21T01:33:55.561234 #23341] DEBUG -- : Adding dependency on rubyzip '~> 2.3.2'
D, [2024-02-21T01:33:55.561239 #23341] DEBUG -- : Adding dependency on sawyer '~> 0.8.2'
D, [2024-02-21T01:33:55.561253 #23341] DEBUG -- : Adding dependency on simplecov '~> 0.18.5'
D, [2024-02-21T01:33:55.561259 #23341] DEBUG -- : Adding dependency on simplecov-html '~> 0.12.3'
D, [2024-02-21T01:33:55.561264 #23341] DEBUG -- : Adding dependency on simpleidn '~> 0.2.1'
D, [2024-02-21T01:33:55.561269 #23341] DEBUG -- : Adding dependency on spreadsheet '~> 1.2.9'
D, [2024-02-21T01:33:55.561274 #23341] DEBUG -- : Adding dependency on systemu '~> 2.6.5'
D, [2024-02-21T01:33:55.561289 #23341] DEBUG -- : Adding dependency on topolys '~> 0.6.2'
D, [2024-02-21T01:33:55.561295 #23341] DEBUG -- : Adding dependency on unf '~> 0.1.4'
D, [2024-02-21T01:33:55.561300 #23341] DEBUG -- : Adding dependency on unf_ext '~> 0.0.9.1'
D, [2024-02-21T01:33:55.561306 #23341] DEBUG -- : Adding dependency on unicode-display_width '~> 2.5.0'
D, [2024-02-21T01:33:55.561311 #23341] DEBUG -- : Adding dependency on uuid '~> 2.3.9'
D, [2024-02-21T01:33:55.561317 #23341] DEBUG -- : Adding dependency on yamler '~> 0.1.0'
D, [2024-02-21T01:33:55.561322 #23341] DEBUG -- : Adding dependency on zliby '~> 0.0.5'
D, [2024-02-21T01:33:55.581161 #23341] DEBUG -- : Processing 61 activation requests
D, [2024-02-21T01:33:55.582464 #23341] DEBUG -- : Activating gem :/ruby/2.7.0/specifications/bundler-2.1.4.gemspec
I, [2024-02-21T01:33:55.972033 #23341]  INFO -- : without_groups = native_ext
I, [2024-02-21T01:33:55.972057 #23341]  INFO -- : g = default
I, [2024-02-21T01:33:55.972063 #23341]  INFO -- : Bundling with groups [default]
Error executing argv: ["--verbose", "--bundle", "Gemfile", "--bundle_path", "/Users/julien/Software/Others/OS-build-release/test-bundle/.bundle", "--bundle_without", "native_ext", "gem_list"]
Error: Could not find racc-1.7.3 in any of the sources in :/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:86:in `block in materialize'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `map!'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `materialize'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/definition.rb:170:in `specs'
:/ruby/2.7.0/gems/bundler-2.1.4/lib/bundler/definition.rb:237:in `specs_for'
:/openstudio_cli.rb:654:in `parse_main_args'
:/openstudio_cli.rb:784:in `execute'
:/openstudio_cli.rb:1987:in `<main>'
eval:195:in `eval'
eval:195:in `require_embedded_absolute'
eval:180:in `block in require_embedded'
eval:174:in `each'
eval:174:in `require_embedded'
eval:133:in `require'
eval:3:in `<main>'
```

@kbenne any clue?

File: ruby/engine/embedded_help.rb [hunk 1]
```diff
@@ -88,7 +88,8 @@ module Kernel
     'jaro_winkler_ext' => 'init_jaro_winkler_ext',\
     'pycall.so' => 'init_pycall',\
     'pycall.dll' => 'init_pycall',\
-    'msgpack/msgpack' => 'init_msgpack'
+    'msgpack/msgpack' => 'init_msgpack',\
+    'racc/cparse' => 'init_cparse'
```
Comment (jmarrec) [inline ruby/engine/embedded_help.rb:92]: I've tried just "racc" too.

Comment (jmarrec) [reply ruby/engine/embedded_help.rb:92]: https://github.com/ruby/racc/blob/e17daedcf3de42b6f4418c39cef166a800033b0e/lib/racc/parser.rb#L195C23-L195C29

**Comment by wenyikuang** [comment]
ubuntu incremental build is back

**Comment by jmarrec** [comment]
With ruby3 gems update, racc is not longer a dependency we have (one of the gems pinned parser to a version before it was introduced)

Also, the above issue is due to bundler thinking the extensions are not built. cf https://github.com/NREL/OpenStudio/commit/1bac721b76232f2c3a70f968b8f79fa3f921c70b

Anyways, closing this PR for now

**Pull request closed without merging**